src/main.cpp

```diff
@@ -4984,7 +5019,7 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
         }
 
         // Disconnect if we connected to ourself
-        if (nNonce == nLocalHostNonce && nNonce > 1)
+        if (pfrom->fInbound && !connman.CheckIncomingNonce(nNonce))
```

Comment:
The code isn't as simple as it used to be - what was wrong with "nNonce == nLocalHostNonce"? it was so much clearer what the code was doing before...